src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -336,18 +336,22 @@ def get_build_options(
             yield from target[field_type].options()
         elif issubclass(field_type, DockerBuildOptionFlagFieldMixin):
             yield from target[field_type].options()
-        elif issubclass(field_type, DockerImageBuildImageCacheToField) or issubclass(
-            field_type, DockerImageBuildImageCacheFromField
-        ):
-            source = InterpolationContext.TextSource(
-                address=target.address, target_alias=target.alias, field_alias=field_type.alias
-            )
-            format = partial(
-                context.interpolation_context.format,
-                source=source,
-                error_cls=DockerImageOptionValueError,
-            )
-            yield from target[field_type].options(format)
+        elif issubclass(field_type, DockerBuildKitOptionField):
```

Comment:
sorry, but I have an itch to scratch here. I think I may have an idea how to further improve this logic slightly.

There are two things going on here now, one is validating option use and the other is to produce the option values themselves.

Looking at this elif chain, I fear it'll be too easy to add a new option that inherits from both `DockerBuildOptionFieldMixin` and `DockerBuildKitOptionField` and then you don't get the behaviour we want.

How about splitting this into two elif chains. The first for option validation (such as the use buildx toggle for buildkit options) and the other for producing the option values. That would also get rid of the duplication we have now.

Reply:
Seems a lot cleaner now - let me know